### PR TITLE
Feature/drc 962 infra setup codecov for recce repo

### PR DESCRIPTION
**Pull request by kentwelcome**
**PR checklist**

- [x] Ensure you have added or ran the appropriate tests for your PR.
- [x] DCO signed

**What type of PR is this?**
Infra

**What this PR does / why we need it**:
- Setup python unit test with code coverage
- Manage the code coverage by Coderov

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -58,3 +58,19 @@ def test_cmd_server_with_cloud(self, mock_state_loader_class, mock_run, mock_get
...
+
+    @patch.object(RecceContext, 'verify_required_artifacts')
+    @patch('recce.cli.asyncio.run')
+    def test_cmd_run(self, mock_asyncio_run, mock_verify_required_artifacts):
+        from recce.cli import run
+
+        mock_verify_required_artifacts.return_value = True, None
+
+        self.runner.invoke(run, [])
+        mock_asyncio_run.assert_called_once()
```
Comment (wcchang1115) [inline tests/test_cli.py]: Hi Kent,
I found that mocking `asyncio.run` led to the warning message:
`sys:1: RuntimeWarning: coroutine 'cli_run' was never awaited`

should we change to `@patch('recce.cli.cli_run')` to check if cli run command is called once?

**Review by wcchang1115 — CHANGES_REQUESTED** [review]

File: tests/test_cli.py (hunk 1 above)
Comment (kentwelcome) [reply tests/test_cli.py]: Yes! You are right!
Changed to mock `cli_run` instead of mocking `asyncio.run`

Comment (kentwelcome) [reply tests/test_cli.py]: Yes! You are right!
Changed to mock cli_run instead of mocking asyncio.run

**Review by kentwelcome — COMMENTED** [review]
Code changed

**Review by wcchang1115 — APPROVED** [review]
Thanks and approve!

**Pull request merged**